uber/models.py

```diff
@@ -630,9 +630,22 @@ def everything(self, location=None):
                                         .order_by(Attendee.full_name).all()
                          if c.AT_THE_CON or not location or int(location) in a.assigned_depts_ints]
             for job in jobs:
-                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted]
+                job._available_staffers = [a for a in attendees if not job.restricted or a.trusted_in(job.location)]
             return jobs, shifts, attendees
 
+        def staffers_by_job_options(self, job):
+            """
+            Format staffers_by_job() for use with the {% options %} template decorator
+            """
+            return [(a.id, a.full_name) for a in self.staffers_by_job(job)]
+
+        def staffers_by_job(self, job):
+            return self.query(Attendee)\
+                .filter_by(staffing=True)\
+                .filter(Attendee.assigned_depts.contains(str(job.location)))\
+                .filter(*(Attendee.trusted_depts.contains(str(job.location)),) if job.restricted else ())\
+                .order_by(Attendee.full_name).all()
+
         def search(self, text, *filters):
```

Comment:
ok further update:   I think they should be refactored into two properties that call the same query because they're 95% similar.

furthermore, the refactored version from staffers_by_job actually does mostly the same thing as available_staffers (almost), but staffers_by_job does all of the filtering in the DB (which is good) while available_staffers does all of the filtering in python (which is bad, it literally gets EVERY attendee and then does filtering with the list comprehension).

I started writing it out and here's what it looks like:

    """
    available_staffers: return a list of ATTENDEES (not just staffers) who:
    1) are assigned to this job's location
    2) are cleared to work this job (they're trusted in this location or the job is unrestricted)
    3) whose existing hours are not overlapped with this job
    """

 """
    staffers_by_job: return a list of attendees who:
    0) have staffing=True
    1) are assigned to this job's location
    2) are cleared to work this job  (they're trusted in this location or the job is unrestricted)
    3) who may or may not have existing hours which overlap with this job
    """

what I think i'll do is create these two as properties on Job (and probably rename them in the end), and have them reference an internal common function which builds the common parts of their query.  each of the two properties will take the query object and then add their own special sauce filtering/ordering on it.

this should have the side effect of making available_staffers no longer need to be cached since I presume it's marked that way because it was inefficient.